apps/webapp/src/components/swap/asset-out-box.tsx

```diff
@@ -16,17 +16,26 @@ import {
 import { ValueViewComponent } from '@penumbra-zone/ui/components/ui/tx/view/value';
 import { groupByAsset } from '../../fetchers/balances/by-asset';
 import { cn } from '@penumbra-zone/ui/lib/utils';
+import { Amount } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/num/v1/num_pb';
 
 const findMatchingBalance = (
-  denom: Metadata | undefined,
+  metadata: Metadata | undefined,
   balances: AssetBalance[],
 ): ValueView | undefined => {
-  if (!denom?.penumbraAssetId) return undefined;
+  if (!metadata?.penumbraAssetId) return undefined;
 
-  return balances.reduce(groupByAsset, []).find(v => {
+  const foundMatch = balances.reduce(groupByAsset, []).find(v => {
     if (v.valueView.case !== 'knownAssetId') return false;
-    return v.valueView.value.metadata?.penumbraAssetId?.equals(denom.penumbraAssetId);
+    return v.valueView.value.metadata?.penumbraAssetId?.equals(metadata.penumbraAssetId);
   });
+
+  if (!foundMatch) {
+    return new ValueView({
+      valueView: { case: 'knownAssetId', value: { metadata, amount: new Amount() } },
+    });
+  }
```

Comment:
Allows UI to show a zero balance in wallet on swap page: 

<img width="185" alt="Screenshot 2024-02-28 at 5 52 51 PM" src="https://github.com/penumbra-zone/web/assets/16624263/df3baadd-5097-4bda-afee-28b703506cc5">